<p align="center">
  <img src="img/GMITLOGO.jpg" width="500"/></p>

# GMIT, H.Dip in Data Analytics, Fundamentals of Data Analysis Project 2018

## 1. Introduction

This repository contains all of the files pertaining to my 2018 project submission for the Fundamentals of Data 
Analysis module of the GMIT H.Dip program in Data Analytics. All of the work within this repository was carried out over the course of a 4 week period in November and December 2018. This Jupyter notebook [1] contains the complete documentation for the project.

### 1.1 Project Objective

The objective of this project is to investigte and explain boxplots and their uses, by researching the subject and then demonstrating the use of box plots using `matplotlib.pyplot`[2]. There are 4 main tasks to carried out [3]:
* Summarise the history of the box plot and situations in which it used.
* Demonstrate the use of the box plot using data of your choosing.
* Explain any relevant terminology such as the terms quartile and percentile. 
* Compare the box plot to alternatives.

### 1.2 Plan for the project

The plan for this project is as follows:
1. Section 2 will give an a brief introduction to boxplots, looking at the history and the use cases. The theory behid them and technical terminology will be explained at this stage
1. In Section 3 I will analyse a data set using boxplots, (insert data set here!)
1. In Section 4 I will discuss some of the findings from Section 3, looking at alternatives to the boxplot, comparing and contrasting the relative strenghts of each approach
1. The project will be concluded in Section 5

## 2. Overview of the Boxplots

### 2.1 Descriptive Statistics

When a statistician or data analyst want to understand and compare datasets, they must have some way way to summarise the dataset. For a one dimensional data set, the data summary should have some idea of where the middle of the dataset lies, and also some measure of what the spread or distribution around this middle value is. A common measure for the middle of a dataset is the *mean*. The mean is calculated by summing the values in the dataset and diving by the number of items in the dataset [4]. The *standard deviation* gives an idea of the spread of the values in the set. The standard deviation is calculated by [5]:
1. Subtracting each value in the data set from the mean
1. Squaring each result
1. Taking the average of these resultant squares
1. Taking the square root of this value

The standard deviation is in the same units as the original data.

A demonstration of how to calculate the mean and standard deviation is shown below:

In [13]:
# Create a list of data
data = [1,2,2,4,5,5,6,7,9,30]

# Calculate mean
total = 0
for i in data:
    total = total + i 
mean = total/len(data)    

# Calculate standard deviation
devsq = 0
for i in data:
    devsq = devsq + (mean - i)**2 
std_dev = (devsq/len(data))**0.5   

print("The mean of the data is : " + str(round(mean,2)))
print("The standard deviation of the data is : " + str(round(std_dev,2)))

636.9
The mean of the data is : 7.1
The standard deviation of the data is : 7.98


This demonstrates how to calculate the mean and standard deviation, note however that in this case the mean is 7.1 and only 2 out of the 10 points in the data set are above this value, the other 8 are below it. The mean has been influenced by the extremely high value (30) in the set.

While the mean and standard deviation are useful for describing a dataset - they can be heavily influenced in data has *outliers* [6]. Outliers are extreme values that deviate from other observations in data [7]. If a dataset has outliers, the *five number summary summary* can be used to better describe it.

### 2.2 The five number summary

### History
### Uses

## 3. How to use the box plot
### Code samples to show how it can be used

## 4. Alternatives
### Comparisons between box plots and other such as histograms

## 6. Conclusion

## References

1. [1] Project Jupyter. Project Jupyter Home. (_https://www.jupyter.org/_)  
1. [2] Matplotlib Development Team. pyplot. (*https://matplotlib.org/api/pyplot_api.html*)
1. [3] Dr. Ian McLoughlin. GMIT. Project 2018: Fundamentals of Data Analysis (_https://github.com/ianmcloughlin/funda-project-2018/raw/master/project.pdf_) 
1. [4] www.mathisfun.com. How to find the mean. (*https://www.mathsisfun.com/mean.html*)
1. [5] www.mathisfun.com. Standard Deviation and Variance. (*https://www.mathsisfun.com/standard-deviation.html*) 
1. [6] www;thoughtco.com What is the five number summary? (https://www.thoughtco.com/what-is-the-five-number-summary-3126237**)
1. [7] www.towardsdatascience.com. A Brief Overview of Outlier Detection Techniques. (https://towardsdatascience.com/a-brief-overview-of-outlier-detection-techniques-1e0b2c19e561)